commandapi-platforms/commandapi-bukkit/commandapi-bukkit-core/src/main/java/dev/jorel/commandapi/CommandAPIBukkitConfig.java

```diff
@@ -0,0 +1,42 @@
+package dev.jorel.commandapi;
+
+import io.papermc.paper.event.server.ServerResourcesReloadedEvent;
+import org.bukkit.plugin.java.JavaPlugin;
+
+/**
+ * A class that contains information needed to configure the CommandAPI on Bukkit-based servers.
+ */
+public class CommandAPIBukkitConfig extends CommandAPIConfig<CommandAPIBukkitConfig> {
+	JavaPlugin plugin;
+	// Default configuration
+	boolean shouldHookPaperReload = true;
+
+	/**
+	 * Creates a new CommandAPIBukkitConfig object. Variables in this
+	 * constructor are required to load the CommandAPI on Bukkit properly.
+	 *
+	 * @param plugin The {@link JavaPlugin} that is loading the CommandAPI
+	 *               This is used when registering events.
+	 */
+	public CommandAPIBukkitConfig(JavaPlugin plugin) {
+		this.plugin = plugin;
+	}
+
+	/**
+	 * Sets the CommandAPI to hook into Paper's {@link ServerResourcesReloadedEvent} when available
+	 * if true. This helps CommandAPI commands to work in datapacks after {@code /minecraft:reload}
+	 * is run.
+	 *
+	 * @param hooked whether the CommandAPI should hook into Paper's {@link ServerResourcesReloadedEvent}
+	 * @return this CommandAPIBukkitConfig
+	 */
+	public CommandAPIBukkitConfig shouldHookPaperReload(boolean hooked) {
+		this.shouldHookPaperReload = hooked;
```

Comment:
This should error if `ServerResourcesReloadedEvent` is not present (i.e. you're running on a Spigot/CraftBukkit server). I think erroring here would be best because this is where users directly declare that they're using this hook.

Reply:
The functionality before this PR was to have `hook-paper-reload` set to `true` by default. This gets checked by `checkDependencies` [here](https://github.com/willkroboth/CommandAPI/blob/dev/dev/commandapi-platforms/commandapi-bukkit/commandapi-bukkit-core/src/main/java/dev/jorel/commandapi/CommandAPIBukkit.java#L145-L154) and outputs a warning to the console if it doesn't hook correctly. As such, we don't need an assertion here.